Helper/Data.php

```diff
@@ -263,20 +263,34 @@ public function cronSubscribeAll($list)
      */
     public function autoResponderAPiEmail($_data, $emailProduct)
     {
+        // send data to autoresponder limit 10 products
         $autoRespId = $this->getGeneralConfig('ac_ar_id');
-        $prod = @$emailProduct[0];
-
-        $address = [
-            'email' => $_data['email'],
-            'name' => $_data['customer_name'],
-            'abandoned_cart_url' => $this->getGeneralConfig('carturl'),
-        ];
         $response = false;
-        if (!empty($prod)) {
-            foreach ($prod as $field => $val) {
-                $address['product_' . $field] = $val;
+        if (!empty($emailProduct) && !empty($_data)) {
+            $address = [
+                'email' => $_data['email'],
+                'name' => $_data['customer_name'],
+                'abandoned_cart_url' => $this->getGeneralConfig('carturl'),
+            ];
+            //If more than one product in abandoned cart iterate to products array
+            if (count($emailProduct)>10) {
+                $address['over_10_products'] = true;
+            } elseif (count($emailProduct) >1) {
+                $length = count($emailProduct);
+                if ($length > 10) {
+                    $length = 10;
+                }
+                for ($i=0; $i < $length; $i++) {
+                    foreach ($emailProduct[$i] as $key => $value) {
+                        $itemNumber = $i + 1;
+                        $address[$key . '_' . $itemNumber] = $value;
+                    }
+                }
+            } else {
+                foreach ($emailProduct[0] as $key => $val) {
+                    $address[$key] = $val;
```

Comment:
Use same `$key` structure (e.g. `product_name_0`) for single product as you are using for multiple products. It makes building the template easier.

Reply:
Added same structure `product_name_1`
https://github.com/sendsmaily/smaily-magento-extension/pull/12/commits/0cfb9706c277c4d2887914376049a2106ad5dbe3